### PR TITLE
Remove tuf target expiration errors from the logs.

### PR DESCRIPTION
**Pull request by cagedmantis**
-Valid targets with valid timestamps are only available for a period after the new agent package has been deployed. When an valid agent that has an expired timestamp is encountered it should not create an error message. This log entry created by this condition has been modified.

### CODE REVIEW

File: cmd/do-agent/main.go [hunk 1]
```diff
@@ -166,20 +155,52 @@ func main() {
...
-			updateAgent(updater)
+			updateAgentWithRestart(updater)
 		}
 	}
 }
 
-func updateAgent(updater update.Updater) {
+// updateAgentWithRestart looks for any available updates to the agent. If an update is found, it will
+// update the agent binary and reinitialize itself. If an update isn't found or fails, it will
+// only log the results of it's attempt.
```
Comment (phillbaker) [inline cmd/do-agent/main.go]: `its`

File: cmd/do-agent/main.go [hunk 2]
```diff
@@ -166,20 +155,52 @@ func main() {
...
+			os.Exit(0)
+		}
+
+		if err == update.ErrUnableToRetrieveTargets {
+			log.Info("No target available for update")
+			os.Exit(0)
+		}
+
+		log.Errorf("Unable to update do-agent: %s\n", err)
+		os.Exit(0)
```
Comment (phillbaker) [inline cmd/do-agent/main.go]: Should we exit(1) here?

File: cmd/do-agent/main.go (hunk 1 above)
Comment (cagedmantis) [inline cmd/do-agent/main.go]: Good catch.

File: cmd/do-agent/main.go (hunk 2 above)
Comment (cagedmantis) [inline cmd/do-agent/main.go]: Seems logical to me.

**Pull request merged**